pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -338,11 +338,16 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	// TODO(random-liu): Remove this after cluster logging supports CRI container log path.
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
-		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
-			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+		if fDest, err := osInterface.Readlink(logSymlink); err != nil {
+			// Remove when the container path do not exist
+			if _, err := osInterface.Stat(osInterface.Dir(fDest)); os.IsNotExist(err) {
```

Comment:
It looks like `os.Readlink` always returns an empty destination when there is an error: https://golang.org/src/os/file_posix.go?s=451:493#L8

So `fDest` will always be `""` here.

Reply:
@tallclair 
I tested it, `os.Readlink` could read a broken sym link and not return error, so it can be used here.

Reply:
Right, but I mean if it does return an error (which is your condition), then fDest will be nil.